p2p/host.go

```diff
@@ -82,112 +113,172 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 		noMdns:      cfg.NoMdns,
+		peerStatus:  peerStatus, // TODO: Store peer status in peer metadata
 	}, nil
 
 }
 
-// bootstrap connects to all provided bootnodes unless noBootstrap is true
+// `bootstrap` connects the host to the defined bootnodes
 func (h *host) bootstrap() {
 	if len(h.bootnodes) == 0 && !h.noBootstrap {
-		log.Debug("no peers to bootstrap to")
+		log.Error(
+			"bootstrap",
+			"error", "NoBootrap must be true if no bootnodes are defined",
+		)
 	}
-
 	for _, peerInfo := range h.bootnodes {
-		log.Debug("Attempting to dial peer", "id", peerInfo.ID, "addrs", peerInfo.Addrs)
+		log.Debug(
+			"bootstrap",
+			"host", h.h.ID(),
+			"peer", peerInfo.ID,
+		)
 		err := h.h.Connect(context.Background(), peerInfo)
 		if err != nil {
-			log.Debug("failed to dial bootstrap peer", "err", err)
+			log.Error("bootstrap", "error", err)
 		}
 	}
 }
 
+// `startMdns` starts the mdns service
 func (h *host) startMdns() {
 	if !h.noMdns {
-		mdns, err := discovery.NewMdnsService(h.ctx, h.h, mdnsPeriod, string(h.protocolId))
+		mdns, err := discovery.NewMdnsService(
+			h.ctx,
+			h.h,
+			mdnsPeriod,
+			string(h.protocolId),
+		)
 		if err != nil {
-			log.Error("error starting MDNS", "err", err)
+			log.Error("start mdns", "error", err)
 		}
 
+		log.Debug(
+			"start mdns",
+			"host", h.h.ID(),
+			"period", mdnsPeriod,
+			"protocol", h.protocolId,
+		)
+
 		mdns.RegisterNotifee(Notifee{ctx: h.ctx, host: h.h})
 
 		h.mdns = mdns
 	}
 }
 
+// `logAddrs` logs the listening addresses of the host
 func (h *host) logAddrs() {
-	addrs := h.h.Addrs()
-	log.Info("You can be reached on the following addresses:")
-	for _, addr := range addrs {
-		fmt.Println("\t" + addr.Encapsulate(h.hostAddr).String())
+	fmt.Println("Listening on the following addresses...")
+	for _, addr := range h.h.Addrs() {
+		fmt.Println(addr.Encapsulate(h.hostAddr).String())
 	}
 }
 
+// `registerStreamHandler` registers the stream handler (see `handleStream`)
 func (h *host) registerStreamHandler(handler func(net.Stream)) {
 	h.h.SetStreamHandler(h.protocolId, handler)
 }
 
+// `connect` connects the host to a specific peer address
 func (h *host) connect(addrInfo peer.AddrInfo) (err error) {
 	err = h.h.Connect(h.ctx, addrInfo)
 	return err
 }
 
-// getExistingStream gets an existing stream for a peer that uses the host's protocolId
-func (h *host) getExistingStream(p peer.ID) net.Stream {
-	conns := h.h.Network().ConnsToPeer(p)
-	for _, conn := range conns {
-		streams := conn.GetStreams()
-		for _, stream := range streams {
-			if stream.Protocol() == h.protocolId {
-				return stream
-			}
-		}
-	}
+// `getStream` attempts to get an existing stream (using `getExistingStream`)
+// and opens a new stream if an existing stream does not exist.
+func (h *host) getStream(p peer.ID) (stream net.Stream, err error) {
+	// stream = h.getExistingStream(p)
 
-	return nil
+	// if stream == nil {
+	stream, err = h.h.NewStream(h.ctx, p, h.protocolId)
```

Comment:
this no longer returns an existing stream

Reply:
That is true. Using an existing stream was preventing both nodes from receiving the status message (both would write to the stream but only one would receive the status message).

I commented this out rather than deleting it because I think we would want to use an existing stream if available but we might need a condition to determine when. I will take another look and either figure out what condition we need or open an issue and update the comment.

Reply:
Ok, updated implementation in `send` method and opened #437